src/stream-tcp-reassemble.c

```diff
@@ -893,10 +894,16 @@ static void GetAppBuffer(TcpStream *stream, const uint8_t **data, uint32_t *data
     }
 }
 
-static inline bool CheckGap(TcpStream *stream, Packet *p)
+/** \internal
+ *  \brief check to see if we should declare a GAP
+ *  Call this when the app layer didn't get data at the requested
+ *  offset.
+ */
+static inline uint32_t CheckGap(TcpSession *ssn, TcpStream *stream, Packet *p)
 {
     const uint64_t app_progress = STREAM_APP_PROGRESS(stream);
     uint64_t last_ack_abs = STREAM_BASE_OFFSET(stream);
+    uint32_t gap = app_progress - last_ack_abs;
```

Comment:
I'm not convinced this is correct. last_ack_abs indicates the right edge of what is ack'd. This may include both gaps and segment data.

When this function is called the only thing we know for sure is that there was no data at the exact offset of 'app_progress'.  One byte later we may have data.

We can have something like:
<pre>
[gap][data][gap][data]
</pre>
all in the window between app_progress and last_ack_abs